### PR TITLE
Expose CPU/Memory charts

### PR DESCRIPTION
**Pull request by fkorotkov**
Fixes https://github.com/cirruslabs/cirrus-ci-docs/issues/93

Here is a screenshot:

image

### CODE REVIEW

File: src/components/common/TaskExecutionInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+      </div>
+    );
+  }
+
+  renderMemoryChart() {
+    let { task } = this.props;
+    let info = task.executionInfo;
+    if (!info.memoryChart) return null;
+    if (info.memoryChart.points.length < 2) return null;
+    let chartPoints = Array(info.memoryChart.points.length);
```
Comment (RDIL) [inline src/components/common/TaskExecutionInfo.tsx:88]:
```suggestion
    let chartPoints = info.memoryChart.points.length;
```
I think this wrapper is redundant?

File: src/components/common/TaskExecutionInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            <Tooltip labelFormatter={index => `Time: ${chartPoints[index].TimestampLabel}`} />
+          </AreaChart>
+        </ResponsiveContainer>
+      </div>
+    );
+  }
+
+  renderMemoryChart() {
+    let { task } = this.props;
+    let info = task.executionInfo;
```
Comment (RDIL) [inline src/components/common/TaskExecutionInfo.tsx:85]:
```suggestion
    let info = task.executionInfo;

```
little bit more space for readability

File: src/components/common/TaskExecutionInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    if (!info.memoryChart) return null;
+    if (info.memoryChart.points.length < 2) return null;
```
Comment (RDIL) [inline src/components/common/TaskExecutionInfo.tsx:87]:
```suggestion
    if (info.memoryChart.points.length < 2) return null;

```
same here

File: src/components/common/TaskExecutionInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+          'Requested Memory': task.instanceResources.memory,
+          'Used Memory': point.value,
+          TimestampLabel: formatDuration(point.secondsFromStart),
+        };
+      }
+    });
+    return (
+      <div className="col">
+        <Typography variant="h6" align="center" className="align-middle">
+          Memory Usage Chart ({memoryUnit})
```
Comment (RDIL) [inline src/components/common/TaskExecutionInfo.tsx:108]:
```suggestion
          Memory Usage ({memoryUnit})
```
saying it is a chart is redundant

File: src/components/common/TaskExecutionInfo.tsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+      chartPoints[index] = {
+        'Requested CPUs': task.instanceResources.cpu,
+        'Used CPUs': point.value,
+        TimestampLabel: formatDuration(point.secondsFromStart),
+      };
+    });
+    return (
+      <div className="col">
+        <Typography variant="h6" align="center" className="align-middle">
+          CPU Usage Chart
```
Comment (RDIL) [inline src/components/common/TaskExecutionInfo.tsx:64]:
```suggestion
          CPU Usage
```

File: src/components/common/TaskExecutionInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+          {this.renderCPUChart()}
+          {this.renderMemoryChart()}
+        </div>
+      </div>
+    );
+  }
+
+  renderCPUChart() {
+    let { task } = this.props;
+    let info = task.executionInfo;
```
Comment (RDIL) [inline src/components/common/TaskExecutionInfo.tsx:50]:
```suggestion
    let info = task.executionInfo;

```
see below for reasoning

File: src/components/common/TaskExecutionInfo.tsx [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    if (!info.cpuChart) return null;
+    if (info.cpuChart.points.length < 2) return null;
+    let chartPoints = Array(info.cpuChart.points.length);
```
Comment (RDIL) [inline src/components/common/TaskExecutionInfo.tsx:53]:
```suggestion
    let chartPoints = Array(info.cpuChart.points.length);

```

File: src/components/common/TaskExecutionInfo.tsx (hunk 1 above)
Comment (fkorotkov) [reply src/components/common/TaskExecutionInfo.tsx:88]: nope, creating an array of this length

Comment (RDIL) [reply src/components/common/TaskExecutionInfo.tsx:88]: Ohh, I see now.

**Comment by kinke** [comment]
I've just accidentally stumbled upon this in https://cirrus-ci.com/task/5723826866618368. Very cool, thanks! :+1:

**Pull request merged**